operator/src/lib.rs

```diff
@@ -331,22 +352,20 @@ impl ControllerStrategy for KafkaStrategy {
         let cluster_spec: KafkaClusterSpec = context.resource.spec.clone();
 
         let mut eligible_nodes = HashMap::new();
+        let test = cluster_spec
+            .brokers
+            .selectors
+            .iter()
+            .map(|(group_name, selector_config)| RoleGroup {
+                name: group_name.to_string(),
+                selector: selector_config.clone().selector.unwrap(),
```

Comment:
selector_config.selector could be None in theory? I know it does not make sense but this could panic.
It is enough to clone only the selector, not the whole config:
`selector: selector_config.selector.clone().unwrap()`

Reply:
I've included this.